src/System.Net.WebSockets/src/System.Net.WebSockets.csproj

```diff
@@ -18,18 +19,31 @@
     <Compile Include="System\Net\WebSockets\WebSocketMessageType.cs" />
     <Compile Include="System\Net\WebSockets\WebSocketReceiveResult.cs" />
     <Compile Include="System\Net\WebSockets\WebSocketState.cs" />
+    <Compile Include="$(CommonPath)\System\Net\WebSockets\ManagedWebSocket.cs">
```

Comment:
> The plan is now that we won't use this for UAP (or Windows) and so we may want to conditionally compile it for platforms where it's used to avoid unnecessary disk footprint.

ASP.NET needs it, though.  Part of the point here is exposing it for others like ASP.NET to be able to use, including on Windows.

> what is the binary size increase due to adding the implementation?

It moves 30k of IL from System.Net.WebSockets.Client.dll (on Unix) to System.Net.WebSockets.dll (all builds).  It'll also end up removing the same amount from System.Net.HttpListener.dll (on Unix, but presumably for UAP as well?), which will be done in a separate PR when the relevant API is added to enable the server-side.